fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/base/ResourceAmount.java

```diff
@@ -27,4 +30,9 @@
  */
 @ApiStatus.Experimental
 public record ResourceAmount<T> (T resource, long amount) {
+	public static final Codec<ResourceAmount<FluidVariant>> FLUID_VARIANT_CODEC = RecordCodecBuilder.create(instance ->
```

Comment:
This has no business being here, fluid-specific code should go in the `fluid` subpackage of the transfer API. This could probably go in `FluidVariant`.